R-packages/covidcast/R/wrangle.R

```diff
@@ -0,0 +1,295 @@
+    x = x %>% dplyr::mutate(!!varname := fun(value, n = abs(n)))
+  }
+  
+  # Remove value column, restore attributes, and return
+  x = dplyr::select(x, -value)
+  attributes(x) = c(attributes(x), attrs)
+  return(x)
+}
+
+##########
+
+#' Aggregate `covidcast_signal` objects into one data frame
+#'
+#' Aggregates `covidcast_signal` objects into one data frame, in either "wide"
+#' or "long" format. (In "wide" aggregation, only the latest issue from each
+#' data frame is retained, and several columns, including `data_source` and
+#' `signal` are dropped; see details below).   
+#' 
+#' @param x Single `covidcast_signal` data frame, or a list of such data
+#'   frames.
+#' @param dt Vector of shifts to apply to the values in the data frame `x`.
+#'   Positive values are interpreted as shifts forward in time, and negative
+#'   values are shifts backward in time. For example, if `dt = 1`, then the
+#'   values are shifted forward 1 day in time (so, data on June 1 becomes data
+#'   on June 2, and so on); if `dt = 0`, then the values are left as is. When
+#'   `x` is a list of data frames, `dt` can either be a single vector of shifts
+#'   or a list of vectors of shifts, this list having the same length as `x` (in
+#'   order to apply, respectively, the same shifts or a different set of shifts
+#'   to each data frame in `x`).
+#' @param format One of either "wide" or "long". The default is "wide".
+#'
+#' @details This function can be thought of having three use cases. In all three
+#'   cases, the result will be a new data frame in either "wide" or "tall"
+#'   format, depending on `format`.
+#'
+#'   The first use case is to apply time-shifts to the values in a given
+#'   `covidcast_signal` object. In this use case, `x` is a `covidcast_signal`
+#'   data frame and `dt` is a vector of shifts. 
+#'
+#'   The second use case is to bind together, into one data frame, signals that
+#'   are returned by `covidcast_signals()`. In this use case, `x` is a list of
+#'   `covidcast_signal` data frames, and `dt` is `NULL`. 
+#'
+#'   The third use case is a combination of the first two: to bind together
+#'   signals returned by `covidcast_signals()`, and simultaneously, apply
+#'   time-shifts to their values. In this use case, `x` is a list of
+#'   `covidcast_signal` data frames, and `dt` is either a vector of shifts---to
+#'   apply the same shifts for each signal in `x`, or a list of vector of
+#'   shifts---to apply different shifts for each signal in `x`.
+#'
+#' @return Data frame of aggregated signals in "wide" or "long" form, depending 
+#'   on `format`. In "long" form, an extra column `dt` is appended to indicate
+#'   the value of the time-shift. In "wide" form, only the latest issue of data
+#'   is retained, and the columns `data_source`, `signal`, `issue`, `lag`,
+#'   `stderr`, `sample_size` are all dropped from the returned data frame. Each
+#'   unique signal---defined by a combination of data source name, signal name,
```

Comment:
Might be worth noting that these are full joins on everything in the list and not some other join type

Reply:
Thanks, done.